src/object/ispy.cpp

```diff
@@ -86,42 +85,98 @@ Ispy::collision(GameObject& , const CollisionHit& )
 }
 
 void
-Ispy::update(float )
+Ispy::update(float dt_sec)
 {
 
-  if (state == ISPYSTATE_IDLE) {
-    // check if a player has been spotted
+  if (m_state == ISPYSTATE_IDLE)
+  {
+    //Check if a player has been spotted
     Vector eye = m_col.m_bbox.get_middle();
-    if (dir == Direction::LEFT) eye = Vector(m_col.m_bbox.get_left(), m_col.m_bbox.get_middle().y);
-    if (dir == Direction::RIGHT) eye = Vector(m_col.m_bbox.get_right(), m_col.m_bbox.get_middle().y);
-    if (dir == Direction::UP) eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_top());
-    if (dir == Direction::DOWN) eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_bottom());
-
-    if (Sector::get().can_see_player(eye)) {
-      m_sprite->set_action((dir == Direction::DOWN) ? "alert-down" : ((dir == Direction::LEFT) ? "alert-left" : "alert-right"), 1);
-      state = ISPYSTATE_ALERT;
+
+    switch (m_dir)
+    {
+      case Direction::DOWN:  eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_bottom());   break;
+      case Direction::UP:    eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_top());      break;
+      case Direction::LEFT:  eye = Vector(m_col.m_bbox.get_left(),     m_col.m_bbox.get_middle().y); break;
+      case Direction::RIGHT: eye = Vector(m_col.m_bbox.get_right(),    m_col.m_bbox.get_middle().y); break;
+      default: break;
+    }
+
+    if (Sector::get().can_see_player(eye))
+    {
+      switch (m_dir)
+      {
+        case Direction::DOWN:  m_sprite->set_action("alert-down",  1); break;
+        case Direction::UP:    m_sprite->set_action("alert-up",    1); break;
+        case Direction::LEFT:  m_sprite->set_action("alert-left",  1); break;
+        case Direction::RIGHT: m_sprite->set_action("alert-right", 1); break;
+        default: break;
+      }
+      m_state = ISPYSTATE_ALERT;
     }
   }
-  if (state == ISPYSTATE_ALERT) {
-    if (m_sprite->animation_done()) {
-      m_sprite->set_action((dir == Direction::DOWN) ? "hiding-down" : ((dir == Direction::LEFT) ? "hiding-left" : "hiding-right"), 1);
-      state = ISPYSTATE_HIDING;
-	  
-	  Sector::get().run_script(script, "Ispy");
+  if (m_state == ISPYSTATE_ALERT)
+  {
+    if (m_sprite->animation_done())
+    {
+      switch (m_dir)
+      {
+        case Direction::DOWN:  m_sprite->set_action("hiding-down",  1); break;
+        case Direction::UP:    m_sprite->set_action("hiding-up",    1); break;
+        case Direction::LEFT:  m_sprite->set_action("hiding-left",  1); break;
+        case Direction::RIGHT: m_sprite->set_action("hiding-right", 1); break;
+        default: break;
+      }
```

Comment:
This pattern comes back often in this file; would it be possible to add a private member function to handle this, with the action name in `const std::string&` argument?